rclcpp/include/rclcpp/node.hpp

```diff
@@ -176,13 +186,43 @@ class Node : public std::enable_shared_from_this<Node>
   template<
     typename MessageT, typename Alloc = std::allocator<void>,
     typename PublisherT = ::rclcpp::Publisher<MessageT, Alloc>>
+  [[deprecated]]
```

Comment:
The deprecation warning should read something like:

> "use the `create_publisher(const std::string &, size_t, const PublisherOptions<Alloc> & = PublisherOptions<Alloc>())` signature instead"

This deprecation I think is still good to do, but I would also be ok to drop this one for this pr, and do it in a follow up pr. Because at the same time we should also provide a way to let users specify even more QoS settings. If we just deprecate this, then I don't think you can, for instance, turn on best effort without invoking a deprecate function.

So let's just drop this deprecation from this pr, despite my previous request, and that should also reduce considerably the number of packages you need to touch to merge this.

Reply:
Rephrasing for my own clarity - We will not deprecate the API that does not have a mandatory `qos_history_depth`, so that the following usage type will be ok in this pass.

```
create_subscription(topic, callback)
create_publisher(topic)
```

Meaning remove the `[[deprecated]]` from this func and the matching `create_subscription`

Reply:
Correct!

I'll deprecate it in a different pr, when I also add an option to pass an rclcpp specific QoS struct, which can only be created by providing a history depth, and then I'd feel ok deprecating the signature that takes the rmw QoS struct.